partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RexConverter.kt

```diff
@@ -127,26 +139,101 @@ internal object RexConverter {
             )
-            val op = rexOpLit(node.value)
+            val op = rexOpLit(pValue)
             return rex(cType, op)
         }
 
+        private fun Literal.toPartiQLValue(): PartiQLValue {
+            val lit = this
+            return when (lit.kind().code()) {
+                LiteralKind.NULL -> nullValue()
+                LiteralKind.MISSING -> missingValue()
+                LiteralKind.STRING -> stringValue(lit.stringValue())
+                LiteralKind.BOOLEAN -> boolValue(lit.booleanValue())
+                LiteralKind.NUM_EXACT -> {
+                    // TODO previous behavior inferred decimals with scale = 0 to be a PartiQLValue.IntValue with
+                    //  PType of numeric. Since we're keeping numeric and decimal, need to take another look at
+                    //  whether the literal should have type decimal or numeric.
+                    val dec = lit.bigDecimalValue()
+                    if (dec.scale() == 0) {
+                        intValue(dec.toBigInteger())
+                    } else {
+                        decimalValue(dec)
+                    }
+                }
+                LiteralKind.NUM_INT -> {
+                    val n = lit.numberValue()
+                    // 1st, try parse as int
+                    try {
+                        val v = n.toInt(10)
+                        return int32Value(v)
+                    } catch (ex: NumberFormatException) {
+                        // ignore
+                    }
+
+                    // 2nd, try parse as long
+                    try {
+                        val v = n.toLong(10)
+                        return int64Value(v)
+                    } catch (ex: NumberFormatException) {
+                        // ignore
+                    }
+
+                    // 3rd, try parse as BigInteger
+                    try {
+                        val v = BigInteger(n)
+                        return intValue(v)
+                    } catch (ex: NumberFormatException) {
+                        throw ex
+                    }
+                }
+                LiteralKind.NUM_APPROX -> {
+                    float64Value(lit.numberValue().toDouble())
+                }
+                LiteralKind.TYPED_STRING -> {
+                    val type = this.dataType()
+                    val typedString = this.stringValue()
+                    when (type.code()) {
+                        DataType.DATE -> {
+                            val value = LocalDate.parse(typedString, DateTimeFormatter.ISO_LOCAL_DATE)
+                            val date = DateTimeValue.date(value.year, value.monthValue, value.dayOfMonth)
+                            dateValue(date)
+                        }
+                        DataType.TIME, DataType.TIME_WITH_TIME_ZONE -> {
+                            val time = DateTimeUtils.parseTimeLiteral(typedString)
+                            val precision = type.precision ?: 6
```

Comment:
I believe these are safe to `!!`.

Reply:
`DataType`'s `precision` method returns a nullable integer (null indicates no precision was explicitly specified like `TIME WITH TIME ZONE '12:34:56'`, so I think we still need to check for null w/ the elvis operator.

Reply:
Hmm actually seems like the parser is eagerly inferring the precision from the time/timestamp string when the type precision argument is unspecified -- https://github.com/partiql/partiql-lang-kotlin/blob/v1-ast-remove-partiqlvalue/partiql-parser/src/main/kotlin/org/partiql/parser/internal/PartiQLParserDefault.kt#L2201-L2208. I don't think we should be doing this eager inference in the parser so I'll change.